File: example-js/aoc.mjs
import { readFileSync } from "fs";
import { parseInput, runPartOne, runPartTwo } from "./puzzle.mjs";

const runPuzzle = { part1: runPartOne, part2: runPartTwo }[process.env.part || "part1"];
if (runPuzzle) {
  const filename = `${process.env.mode == "test" ? "input-test" : "input"}.txt`;
  const input = readFileSync(filename, { encoding: "utf8" }).trim().split("\n");
  console.log("Result:", runPuzzle(parseInput(input)));
} else {
  console.log("Unknown part");
}

File: day02/aoc.nim
import std/[os, strutils]
import puzzle

let runPuzzle =
  case getEnv("part"):
    of "part1", "": runPartOne
    of "part2": runPartTwo
    else: quit "Unknown part"

let filename = (if getEnv("mode") == "test": "input-test" else: "input") & ".txt"
echo "Result: ", readFile(filename).strip.splitLines.parseInput.runPuzzle
